### PR TITLE
Add Auth Controller documentation

### PR DESCRIPTION
**Pull request by Shiranuit**
## What does this PR do ?

Add Auth controller documentation and tests

### How should this be manually tested?
Run: `$ docker-compose -f .ci/doc/docker-compose.yml run doc-tests node index`

### CODE REVIEW

File: doc/1/controllers/auth/delete-my-credentials/index.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+code: true
+type: page
+title: DeleteMyCredentialsAsync
+description: Delete the current user's credentials for the specified strategy
+---
+
+# DeleteMyCredentialsAsync
+
+Delete the current user's credentials for the specified strategy. If the credentials that generated the current JWT are removed, the user will remain logged in until he logs out or his session expires, after that they will no longer be able to log in with the deleted credentials.
```
Comment (y-abs) [inline doc/1/controllers/auth/delete-my-credentials/index.md]:
```suggestion
Delete the current user's credentials for the specified strategy. If the credentials that generated the current JWT are removed, the user will remain logged in until he logs out or his session expires. After that they will no longer be able to log in with the deleted credentials.
```

File: doc/1/controllers/auth/refresh-token/index.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+**Optional:**
+
+| Arguments   | Type              | Description                                                                 |
+|-------------|-------------------|-----------------------------------------------------------------------------|
+| `expiresIn` | <pre>string</pre> | Set the expiration duration (default: depends on Kuzzle configuration file) |
+
+## Return
+
+A JObject which has the following properties:
```
Comment (y-abs) [inline doc/1/controllers/auth/refresh-token/index.md]:
```suggestion
A JObject with the following properties:
```

**Review by y-abs — APPROVED** [review]

**Comment by Aschen** [comment]
There is no need to put 2 labels, only one will be used to construct the changelog.
You can just use `changelog:documentation`

File: doc/1/controllers/auth/check-token/index.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+|----------|-------------------|-------------|
+| `token`  | <pre>string</pre> | JWT token   |
+
+## Return
+
+A JObject which has the following properties:
+
+| Property   | Type              | Description                      |
+|------------|-------------------|----------------------------------|
+| valid      | <pre>bool</pre>   | Token validity                   |
```
Comment (Aschen) [inline doc/1/controllers/auth/check-token/index.md]: Property name should be between backtick (this comment apply everywhere)

```suggestion
|  `valid`      | <pre>bool</pre>   | Token validity                   |
```

File: doc/1/controllers/auth/check-token/snippets/check-token.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+try {
+  JObject response = (await kuzzle.Auth.LoginAsync("local", JObject.Parse("{username: 'foo', password: 'bar'}")));
```
Comment (Aschen) [inline doc/1/controllers/auth/check-token/snippets/check-token.cs]: Why parenthesis around the call ?
Also you should wrap line greater than 80 column (apply everywhere)

File: doc/1/controllers/auth/check-token/snippets/check-token.test.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+name: auth#checkToken
```
Comment (Aschen) [inline doc/1/controllers/auth/check-token/snippets/check-token.test.yml]:
```suggestion
name: Auth#checkTokenAsync
```

File: doc/1/controllers/auth/create-my-credentials/snippets/create-my-credentials.test.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+name: auth#CreateMyCredentials
```
Comment (Aschen) [inline doc/1/controllers/auth/create-my-credentials/snippets/create-my-credentials.test.yml]:
```suggestion
name: Auth#CreateMyCredentialsAsync
```

File: doc/1/controllers/auth/credentials-exist/snippets/credentials-exist.test.yml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+name: auth#credentialsExist
```
Comment (Aschen) [inline doc/1/controllers/auth/credentials-exist/snippets/credentials-exist.test.yml]:
```suggestion
name: Auth#CredentialsExistAsync
```

File: doc/1/controllers/auth/delete-my-credentials/snippets/delete-my-credentials.test.yml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+name: auth#deleteMyCredentials
```
Comment (Aschen) [inline doc/1/controllers/auth/delete-my-credentials/snippets/delete-my-credentials.test.yml]:
```suggestion
name: Auth#DeleteMyCredentialsAsync
```

File: doc/1/controllers/auth/get-current-user/snippets/get-current-user.test.yml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+name: auth#getCurrentUser
```
Comment (Aschen) [inline doc/1/controllers/auth/get-current-user/snippets/get-current-user.test.yml]:
```suggestion
name: Auth#getCurrentUserAsync
```

File: doc/1/controllers/auth/get-my-credentials/snippets/get-my-credentials.test.yml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+name: auth#getMyCredentials
```
Comment (Aschen) [inline doc/1/controllers/auth/get-my-credentials/snippets/get-my-credentials.test.yml]:
```suggestion
name: Auth#getMyCredentialsAsync
```

File: doc/1/controllers/auth/get-my-rights/snippets/get-my-rights.test.yml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+name: auth#getMyRights
```
Comment (Aschen) [inline doc/1/controllers/auth/get-my-rights/snippets/get-my-rights.test.yml]:
```suggestion
name: Auth#GetMyRightsAsync
```

File: doc/1/controllers/auth/get-strategies/snippets/get-strategies.test.yml [hunk 12]
```diff
@@ -0,0 +1,7 @@
+name: auth#getStrategies
```
Comment (Aschen) [inline doc/1/controllers/auth/get-strategies/snippets/get-strategies.test.yml]:
```suggestion
name: Auth#GetStrategiesAsync
```

File: doc/1/controllers/auth/login/snippets/login.test.yml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+---
+name: auth#login
```
Comment (Aschen) [inline doc/1/controllers/auth/login/snippets/login.test.yml]:
```suggestion
name: Auth#LoginAsync
```

File: doc/1/controllers/auth/logout/snippets/logout.test.yml [hunk 14]
```diff
@@ -0,0 +1,7 @@
+name: auth#logout
```
Comment (Aschen) [inline doc/1/controllers/auth/logout/snippets/logout.test.yml]:
```suggestion
name: Auth#LogoutAsync
```

File: doc/1/controllers/auth/refresh-token/snippets/refresh-token.test.yml [hunk 15]
```diff
@@ -0,0 +1,7 @@
+name: auth#refreshToken
```
Comment (Aschen) [inline doc/1/controllers/auth/refresh-token/snippets/refresh-token.test.yml]:
```suggestion
name: Auth#RefreshTokenAsync
```

File: doc/1/controllers/auth/update-my-credentials/snippets/update-my-credentials.test.yml [hunk 16]
```diff
@@ -0,0 +1,9 @@
+name: auth#updateMyCredentials
```
Comment (Aschen) [inline doc/1/controllers/auth/update-my-credentials/snippets/update-my-credentials.test.yml]:
```suggestion
name: Auth#UpdateMyCredentialsAsync
```

File: doc/1/controllers/auth/update-self/snippets/update-self.test.yml [hunk 17]
```diff
@@ -0,0 +1,7 @@
+name: auth#updateSelf
```
Comment (Aschen) [inline doc/1/controllers/auth/update-self/snippets/update-self.test.yml]:
```suggestion
name: Auth#UpdateSelfAsync
```

File: doc/1/controllers/auth/validate-my-credentials/snippets/validate-my-credentials.test.yml [hunk 18]
```diff
@@ -0,0 +1,7 @@
+name: auth#validateMyCredentials
```
Comment (Aschen) [inline doc/1/controllers/auth/validate-my-credentials/snippets/validate-my-credentials.test.yml]:
```suggestion
name: Auth#ValidateMyCredentialsAsync
```

**Review by Aschen — CHANGES_REQUESTED** [review]

**Review by Aschen — APPROVED** [review]

**Pull request merged**